04d_deployment_with_solara.ipynb

```diff
@@ -0,0 +1,181 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "Page()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c111793c-9f98-4a7a-9d13-22cad993719e",
+   "metadata": {},
+   "source": [
+    "Note that solara displays the dropdown as an ipywidgets dropdown rather than as an ipyvuetify widget, since we explicitly asked for an ipywidget dropdown."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "73ee9beb-55ee-4a8a-aa4b-877b2cc99122",
+   "metadata": {},
+   "source": [
+    "**EXERCISE:**\n",
+    "\n",
+    "Import the first version of the dashboard we made and make it into a solara app. Call the object you make `Page` like we did above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "322b8f73-232d-4602-aea1-f4ccb3327ef3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from dashboard.main import main_widget\n",
+    "\n",
+    "@solara.component\n",
+    "def Page():\n",
+    "    # FILL THIS IN"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e8b61836-32af-4940-b2b8-3bdb44f2dc9b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "Page()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "050d5ee9-dcfd-40be-ae20-70ae18912ce1",
+   "metadata": {},
+   "source": [
+    "## Deploying to PyCafe\n",
+    "\n",
+    "[PyCafe](https://py.cafe) is a web service that utilizes Pyodide to serve up a web app that runs entirely in your browser. There is no Python kernel running in the background, so it scales well. The service is still in beta so you should expect glitches. It is frequently necessary to use the \"System Restart\" menu option to get `scipy` to launch properly, for example. One other thing to be aware of: PyCafe is running in your broswer, so unless you frequently use the \"Push Changes to Cloud\" menu item any changes you have made are lost if your browser page reloads or if you do a \"System Restart\".\n",
+    "\n",
+    "Two versions of the dashboard in this tutorial have been deployed to PyCafe so that you can take a look at them.\n",
+    "\n",
+    "+ The pure-solara version of the dashboard is [here](https://py.cafe/mwcraig/solara-global-temp-analysis). The only modifcation necessary was to change the `DATA` folder variable from `data` to `.`, since PyCafe does not let you make folders.\n",
+    "+ The initial version of the dashboard is [here](https://py.cafe/mwcraig/plain-widget-2024-dashboard). A couple of modifications were necessary:\n",
+    "    + `widgets.py` was renamed to `tutorial_widgets.py`\n",
+    "    + `main.py` imports from `tutorial_widgets` instead of `dashboard.widgets` because i) folders are not allowed, and ii) `widgets` would conflict with `import ipywidgets as widgets`.\n",
+    "    + The `DATA` folder was changed from `data` to `.`.\n",
+    "+ The `ipyautoui` version of the dashboard is not on PyCafe because `ipyautoui` isn't supported, apparently."
```

Comment:
I just found this snarky, keep if you wish.
```suggestion
    "+ The `ipyautoui` version of the dashboard is not on PyCafe because `ipyautoui` isn't supported (yet?)."
```